File: model/src/main/java/com/testerum/model/step/operation/UpdateComposedStepDef.kt
package com.testerum.model.step.operation

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.testerum.model.infrastructure.path.Path
import com.testerum.model.step.ComposedStepDef

data class UpdateComposedStepDef @JsonCreator constructor(@JsonProperty("oldPath") val oldPath: Path,
                                                          @JsonProperty("composedStepDef") val composedStepDef: ComposedStepDef)

File: web/web-frontend/src/app/model/step/UpdateComposedStepDef.ts
import {Path} from "../infrastructure/path/path.model";
import {ComposedStepDef} from "../composed-step-def.model";
import {JsonUtil} from "../../utils/json.util";
import {Serializable} from "../infrastructure/serializable.model";

export class UpdateComposedStepDef implements Serializable<UpdateComposedStepDef> {

    oldPath: Path;
    composedStepDef: ComposedStepDef;

    constructor(oldPath: Path, composedStepDef: ComposedStepDef) {
        this.oldPath = oldPath;
        this.composedStepDef = composedStepDef;
    }

    deserialize(input: Object): UpdateComposedStepDef {
        throw new Error("Unimplemented method");
    }

    serialize(): string {
        return "" +
            '{' +
            '"oldPath":' + JsonUtil.serializeSerializable(this.oldPath) + ',' +
            '"composedStepDef":' + JsonUtil.serializeSerializable(this.composedStepDef) +
            '}'
    }
}
